main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -1242,6 +1309,75 @@ object Lowering {
     mkTag(Enums.HeadTerm, s"App$arity", innerExp, Types.HeadTerm, loc)
   }
 
+  /**
+    * Make the array of MpmcAdmin objects which will be passed to `selectFrom`
+    */
+  private def mkChannelAdminArray(rs: List[SelectChannelRule], channels: Array[(Symbol.VarSym, Expression)], pur: Type, eff: Type, loc: SourceLocation): Expression = {
```

Comment:
You pass in `pur` and `eff`, but as far as I can tell everything return here should be pure, i.e. we can just use Type.Pure and Type.Empty (I think its called).

Reply:
(In fact its wrong to use `pur` and `eff` if you apply a function that is pure (i.e. `Type.mkPureArrow`).